llvm/lib/Target/Xtensa/AsmParser/XtensaAsmParser.cpp

```diff
@@ -480,6 +485,12 @@ bool XtensaAsmParser::matchAndEmitInstruction(SMLoc IDLoc, unsigned &Opcode,
   case Match_InvalidImm1_16:
     return Error(RefineErrorLoc(IDLoc, Operands, ErrorInfo),
                  "expected immediate in range [1, 16]");
+  case Match_InvalidImm1n_15:
+    return Error(RefineErrorLoc(IDLoc, Operands, ErrorInfo),
+                 "expected immediate in range [-1, 15] except 0");
+  case Match_InvalidImm32n_95:
+    return Error(RefineErrorLoc(IDLoc, Operands, ErrorInfo),
+                 "expected immediate in range [-32, 95] except 0");
```

Comment:
- `isImm1n_15` checks -1..15, except 0 and the error message is reflecting this.
- `isImm32n_95`, as implemented, includes 0, but the error here says that it does not.

Either the implementation, or the error message seems to be incorrect.

Reply:
Thank you very much for comments. I fixed error message.